engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2246,6 +2306,50 @@ public LogicalOperatorInfo getLogicalOperatorInfo(String operatorName)
     return fillLogicalOperatorInfo(operatorMeta);
   }
 
+  public LogicalModuleInfo getLogicalModuleInfo(ModuleMeta module, boolean flatten)
+  {
+    LogicalModuleInfo lmi = new LogicalModuleInfo();
+    lmi.name = module.getName();
+    lmi.className = module.getModule().getClass().getName();
+    if (flatten) {
+      for (OperatorMeta operatorMeta : module.getDag().getAllOperators()) {
+        if (operatorMeta.getModuleName() == null) {
+          String fullPath = module.getFullName() + LogicalPlan.MODULE_NAMESPACE_SEPARATOR + operatorMeta.getName();
+          lmi.operators.add(fillLogicalOperatorInfo(getLogicalPlan().getOperatorMeta(fullPath)));
+        }
+      }
+      for (StreamMeta streamMeta : module.getDag().getAllStreams()) {
```

Comment:
You are going over the same data structure twice. When you add the operator, you already know the streams connected to the output ports.